gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagerUtils.java

```diff
@@ -236,6 +243,13 @@ static void incrementJobAttempt(DagNode<JobExecutionPlan> dagNode) {
     dagNode.getValue().setCurrentAttempts(dagNode.getValue().getCurrentAttempts() + 1);
   }
 
+  /**
+   * Increment the value of {@link JobExecutionPlan#currentGeneration}
```

Comment:
I read the code to confirm the other day... but then forgot to mention during my prior review.

small suggestion: you could leave a comment clarifying this is thread safe by reminding the reader that `Dag`s are consistently assigned to a single `DagManagerThread` on the basis of the execution ID